proposals/0053-responsive-images.md

```diff
@@ -0,0 +1,273 @@
+
+These are a lot of attributes to remember and understand, though the final three have values that are usually safe to think of as dependent on just whether the image is onscreen when the page loads. Astro Image already sets `loading` and `decoding` to `lazy` and `async` by default. However `srcset` and `sizes` have no simple rules because they depend on how the image will be displayed, and can be very hard to do correctly. Images also need to be styled correctly if they are to be responsive and avoid CLS.
+
+# Goals
+
+- A single `layout` prop that, when combined with existing props, sets the attributes that will make an image responsive and following best practices.
+- Config options to change the defaults for all images
+- Backwards-compatible, so that existing images are unaffected unless they set the props or config options.
+- Add support for optional cropping in image services
+- Images displayed correctly, even if image service doesn't support cropping
+
+# Non-goals
+
+- Placeholder support
+- Automatic provider detection
+- Art direction
+- Implementing crop support for all existing image services
+
+# Detailed Design
+
+When a user sets the layout, either via the prop or as a default in the config, the image component will auto-generate defaults for `srcset` and `sizes`. It will also apply styles to determine the resizing behavior. This will be done according to these rules for each layout.
+
+For each of these, `<width>` is a placeholder for the value passed by the user as the image `width` prop. The **widths** value refers to the widths of the images in the `srcset`. The **sizes** value is the string value set as the `sizes` attribute.
+
+### Image `srcset`
+
+The image `srcset` tells the browser which images are available. We want to generate a list of sources that allow the browser to always download an image that gives the best balance of quality and file size. There `srcset` is specified as a list of image candidates and conditions. A condition is either the width of the image, or the pixel density. The best result is almost always achieved by using widths rather than densities. This is because it allows the browser to choose the best size, and it can use any criteria it wants to choose that. For example it can choose a lower resolution image if the device is on a slow or metered network. Conversely, if using a density value, the browser _must_ download the resolution that matches the screen.
+
+For this reason, we will generate a srcset with width conditions, based on the width and layout props.
+
+These are indicative implementations of the functions that generate the image widths for the `srcset`. The `breakpoints` argument is an array of possible screen resolutions, which it uses to choose candidate breakpoints. The default list is shown in the section below. In the real implementation, widths that are larger than the source image would be filtered out.
```

Comment:
> In the real implementation, widths that are larger than the source image would be filtered out.

How does this work for remote images without `inferSize`?

Reply:
It would probably rely on image service options to prevent upscaling. It might request a larger size but it would return the original dimensions.

Reply:
...but for the built-in service I think we'd just not include super large sizes at all by default